com.unity.render-pipelines.universal/Runtime/2D/Light2DManager.cs

```diff
@@ -103,15 +104,31 @@ private static bool ContainsDuplicateGlobalLight(int sortingLayerIndex, int blen
         public static SortingLayer[] GetCachedSortingLayer()
         {
             if (s_SortingLayers is null)
-            {
                 s_SortingLayers = SortingLayer.layers;
-            }
-#if UNITY_EDITOR
-            // we should fix. Make a non allocating version of this
-            if (!Application.isPlaying)
-                s_SortingLayers = SortingLayer.layers;
-#endif
+
             return s_SortingLayers;
         }
+
+#if UNITY_EDITOR
+        // Gets called on the start of the lighting pass
+        public static void InitializeFrame()
+        {
+            UpdateLightsTargetSortingLayer();
+        }
+
+        private static void UpdateLightsTargetSortingLayer()
+        {
+            var layers = SortingLayer.layers;
+            if (GetCachedSortingLayer().Length + 1 == layers.Length)
+            {
+                var layer = layers.Except(s_SortingLayers).FirstOrDefault();
+                foreach (var light in lights)
```

Comment:
not every light is rendered at every frame... Also it doesn't change during runtime...

Reply:
Are not all lights registered in the light2dmanager? This only gets updated in editor.